File: CarRental.Api/Models/BookingDetails.cs
using CarRental.Api.Enums;

namespace CarRental.Api.Models;

public class BookingDetails
{
    public string BookingReferenceNumber { get; set; } = string.Empty;
    public string DriverName { get; set; } = string.Empty;
    public DocumentType DocumentType { get; set; }
    public string DocumentNumber { get; set; } = string.Empty;
    public string PickupLocation { get; set; } = string.Empty;
    public ProviderType Provider { get; set; }
    public ProviderVehicle SelectedVehicle { get; set; } = new();
    public decimal TotalPrice { get; set; }
    public string CancellationPolicy { get; set; } = string.Empty;
    public DateTime PickupDate { get; set; }
    public DateTime ReturnDate { get; set; }
}

File: car-rental-ui/src/app/models/booking-details.ts
import { ProviderVehicle } from './provider-vehicle';

export interface BookingDetails {
  bookingReferenceNumber: string;
  driverName: string;
  provider: string;
  pickupLocation: string;
  pickupDate: string;
  returnDate: string;
  totalPrice: number;
  cancellationPolicy: string;
  documentType: string;
  documentNumber: string;
}